src/Microsoft.AspNetCore.Routing.Abstractions/LinkGeneratorContext.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class LinkGeneratorContext
+    {
+        public Address Address { get; set; }
+
+        public IDictionary<string, object> AmbientValues { get; set; }
+
+        public IDictionary<string, object> SuppliedValues { get; set; }
```

Comment:
Future design note:

Having address on here is pretty strange. You'd be reaching into the FINDER to get an address and then calling the link generator. Basically, using a low lever concept to call into a high level concept.